File: jars/src/data/scripts/weapons/dakka_FlakOnExplosionEffect.java
package data.scripts.weapons;

import com.fs.starfarer.api.Global;
import com.fs.starfarer.api.combat.CombatEngineAPI;
import com.fs.starfarer.api.combat.DamagingProjectileAPI;
import com.fs.starfarer.api.combat.ProximityExplosionEffect;
import com.fs.starfarer.api.util.Misc;
import org.lazywizard.lazylib.MathUtils;
import org.lwjgl.util.vector.Vector2f;

import java.awt.*;

public class dakka_FlakOnExplosionEffect implements ProximityExplosionEffect {
    @Override
    public void onExplosion(DamagingProjectileAPI explosion, DamagingProjectileAPI originalProjectile) {
        CombatEngineAPI engine = Global.getCombatEngine();

        engine.addHitParticle(
                originalProjectile.getLocation(),
                Misc.ZERO,
                100f,
                1f,
                0.1f,
                originalProjectile.getProjectileSpec().getCoreColor()
        );

        for (int i = 0; i < 10; i++) {
            engine.addNebulaParticle(
                    originalProjectile.getLocation(),
                    MathUtils.getRandomPointInCircle(Misc.ZERO,30f),
                    MathUtils.getRandomNumberInRange(25f, 50f),
                    1.5f,
                    0.1f,
                    0.3f,
                    MathUtils.getRandomNumberInRange(2f, 2.5f),
                    new Color(50, 48, 45, 120),
                    true
            );
        }
    }
}

File: jars/src/data/scripts/weapons/dakka_FlakOnExplosionEffect.kt
package data.scripts.weapons

import com.fs.starfarer.api.Global
import com.fs.starfarer.api.combat.DamagingProjectileAPI
import com.fs.starfarer.api.combat.ProximityExplosionEffect
import com.fs.starfarer.api.util.Misc
import org.lazywizard.lazylib.MathUtils
import java.awt.Color

class dakka_FlakOnExplosionEffect : ProximityExplosionEffect {
    override fun onExplosion(explosion: DamagingProjectileAPI, originalProjectile: DamagingProjectileAPI) {
        val engine = Global.getCombatEngine()
        val effectCol = Color(
                originalProjectile.projectileSpec.fringeColor.red,
                originalProjectile.projectileSpec.fringeColor.green,
                originalProjectile.projectileSpec.fringeColor.blue,
                70
        )
        engine.addHitParticle(
                explosion.location,
                Misc.ZERO,
                explosion.collisionRadius * 2f,
                0.8f,
                0.1f,
                effectCol
        )
        engine.addSmoothParticle(
                explosion.location,
                Misc.ZERO,
                explosion.collisionRadius * 3f,
                0.8f,
                0.1f,
                effectCol
        )
        for (i in 0..9) {
            engine.addNebulaParticle(
                    explosion.location,
                    MathUtils.getRandomPointInCircle(Misc.ZERO, 15f),
                    MathUtils.getRandomNumberInRange(explosion.collisionRadius * 0.5f, explosion.collisionRadius * 2f),
                    2f,
                    0f,
                    0.3f,
                    MathUtils.getRandomNumberInRange(1.5f, 3f),
                    Color(22, 21, 20, 140),
                    true
            )
        }
    }
}